game-core/src/main/java/games/strategy/engine/data/annotations/GameProperty.java

```diff
@@ -6,9 +6,8 @@
 import java.lang.annotation.Target;
 
 /**
- * Annotation that marks a method as a setter called through reflection by the GameParser (through the xml) and/or
- * PropertyUtil (through the
- * ChangeFactory).
+ * Annotation that marks a method as a setter called by the GameParser (through the xml) and/or
+ * PropertyUtil (through the ChangeFactory).
```

Comment:
Is this annotation still required?  I thought it was only present to alert someone that the target may be called reflectively.

Reply:
I believe this is still the only case, but I decided to keep it for now for readability, and to give us some info about _what kind_ of setter we're talking about.
Currently there are "adder" setters that modify a field if it has already been initialized, the annotation tells us that.

Reply:
True, but I have a feeling that without `ValidateAttachmentsTest` to enforce its usage and verify that its semantics are correct, we're going to find they become out-of-sync with reality, specifically when new properties are added.

Reply:
Valid point